src/main/java/jenkins/bouncycastle/api/PEMManager.java

```diff
@@ -0,0 +1,394 @@
+
+/**
+ * A class that provides an API to manager PEM format, providing additional methods to handle Keys, Certificates,
+ * Fingerprints, etc The supported algorithms will depend on the underlying version of BouncyCastle
+ *
+ */
+public class PEMManager {
+
+    /**
+     * Stores the internal Bouncy Castle or JCA object
+     */
+    private Object pemObject;
+
+    /**
+     * Creates a {@link PEMManager} by reading a PEM file
+     * 
+     * @param pemFile {@link File} pointing to the PEM file to read
+     * @throws IOException launched if a problem exists reading the PEM information or the {@link File}
+     * @throws BCPasswordException in case PEM is passphrase protected and none or wrong is provided
+     */
+    public PEMManager(@Nonnull File pemFile) throws IOException {
+        decodePEM(pemFile, null);
+    }
+
+    /**
+     * Creates a {@link PEMManager} by reading a PEM file
+     * 
+     * @param pemFile {@link File} pointing to the PEM file to read
+     * @param passphrase passphrase for the encrypted PEM data. null if PEM data is not passphrase protected
+     * @throws IOException launched if a problem exists reading the PEM information or the {@link File}
+     * @throws BCPasswordException in case PEM is passphrase protected and none or wrong is provided
+     */
+    public PEMManager(@Nonnull File pemFile, @Nullable String passphrase) throws IOException {
+        decodePEM(pemFile, passphrase);
+    }
+
+    /**
+     * Creates a {@link PEMManager} by reading PEM formated data from a {@link String}
+     * 
+     * @param pem {@link String} with the PEM data
+     * @throws IOException launched if a problem exists reading the PEM information
+     * @throws BCPasswordException in case PEM is passphrase protected and none or wrong is provided
+     */
+    public PEMManager(@Nonnull String pem) throws IOException {
+        decodePEM(pem, null);
+    }
+
+    /**
+     * Creates a {@link PEMManager} by reading PEM formated data from a {@link String}
+     * 
+     * @param pem {@link String} with the PEM data
+     * @param passphrase passphrase for the encrypted PEM data. null if PEM data is not passphrase protected
+     * @throws IOException launched if a problem exists reading the PEM information
+     * @throws BCPasswordException in case PEM is passphrase protected and none or wrong is provided
+     */
+    public PEMManager(@Nonnull String pem, @Nullable String passphrase) throws IOException {
+        decodePEM(pem, passphrase);
+    }
+
+    /**
+     *
+     * Creates a {@link PEMManager} from an {@link Object} that can be of any supported type by Bouncy Castle or JCA:
+     * {@link Key}, {@link PrivateKey}, {@link KeyPair}, {@link Certificate}, etc.
+     *
+     * @param pemObject object to manage
+     */
+    public PEMManager(@Nonnull Object pemObject) {
+        this.pemObject = pemObject;
+    }
+
+    private void decodePEM(@Nonnull File pemFile, @Nullable String passphrase) throws IOException {
```

Comment:
this should probably be `public static PEMEncodable read(File file, String passphrase) throws IOException, UnrecoverableEntryException`